Sources/SwiftSyntaxBuilder/gyb_generated/DeclBuildables.swift

```diff
@@ -0,0 +1,1082 @@
+    genericWhereClause: GenericWhereClause? = nil,
+    members: MemberDeclBlock
+  ) {
+    self.init(
+      attributes: attributesBuilder(),
+      modifiers: modifiersBuilder(),
+      structKeyword: structKeyword,
+      identifier: SyntaxFactory.makeIdentifier(identifier),
+      genericParameterClause: genericParameterClause,
+      inheritanceClause: inheritanceClause,
+      genericWhereClause: genericWhereClause,
+      members: members
+    )
+  }
+}
+
+extension ProtocolDecl {
+  public init(
+    @AttributeListBuilder attributesBuilder: () -> AttributeList = { .empty },
+    @ModifierListBuilder modifiersBuilder: () -> ModifierList = { .empty },
+    protocolKeyword: TokenSyntax = Tokens.`protocol`,
+    identifier: String,
+    inheritanceClause: TypeInheritanceClause? = nil,
+    genericWhereClause: GenericWhereClause? = nil,
+    members: MemberDeclBlock
+  ) {
+    self.init(
+      attributes: attributesBuilder(),
+      modifiers: modifiersBuilder(),
+      protocolKeyword: protocolKeyword,
+      identifier: SyntaxFactory.makeIdentifier(identifier),
+      inheritanceClause: inheritanceClause,
+      genericWhereClause: genericWhereClause,
+      members: members
+    )
+  }
+}
+
+extension ExtensionDecl {
+  public init(
+    @AttributeListBuilder attributesBuilder: () -> AttributeList = { .empty },
+    @ModifierListBuilder modifiersBuilder: () -> ModifierList = { .empty },
+    extensionKeyword: TokenSyntax = Tokens.`extension`,
+    extendedType: TypeBuildable,
+    inheritanceClause: TypeInheritanceClause? = nil,
+    genericWhereClause: GenericWhereClause? = nil,
+    members: MemberDeclBlock
+  ) {
+    self.init(
+      attributes: attributesBuilder(),
+      modifiers: modifiersBuilder(),
+      extensionKeyword: extensionKeyword,
+      extendedType: extendedType,
+      inheritanceClause: inheritanceClause,
+      genericWhereClause: genericWhereClause,
+      members: members
+    )
+  }
+}
+
+extension MemberDeclBlock {
+  public init(
+    leftBrace: TokenSyntax = Tokens.`leftBrace`,
+    @MemberDeclListBuilder membersBuilder: () -> MemberDeclList = { .empty },
+    rightBrace: TokenSyntax = Tokens.`rightBrace`
+  ) {
+    self.init(
+      leftBrace: leftBrace,
+      members: membersBuilder(),
+      rightBrace: rightBrace
+    )
+  }
+}
+
+extension SourceFile {
+  public init(
+    @CodeBlockItemListBuilder statementsBuilder: () -> CodeBlockItemList = { .empty },
+    eofToken: TokenSyntax
+  ) {
+    self.init(
+      statements: statementsBuilder(),
+      eofToken: eofToken
+    )
+  }
+}
+
+extension FunctionParameter {
+  public init(
+    @AttributeListBuilder attributesBuilder: () -> AttributeList = { .empty },
+    firstName: TokenSyntax? = nil,
+    secondName: TokenSyntax? = nil,
+    colon: TokenSyntax? = Tokens.`colon`,
+    type: TypeBuildable? = nil,
+    ellipsis: TokenSyntax? = Tokens.`ellipsis`,
+    defaultArgument: InitializerClause? = nil,
+    trailingComma: TokenSyntax? = Tokens.`comma`
```

Comment:
What should we do with optional parameters? 
As I see, this could be a problem. Because here we are adding a comma. But the last parameter should not have one. 
I don't even think the generated code would compile?

Reply:
Hmm. Maybe it’d be better to not default the token in that case. That way the user needs to be explicit whether he wants it or not.